### ISSUE TITLE
rationalize/numerator/denominator ineffective without prior factoring on input like 1/2*x+3/7

### ISSUE THREAD

**Issue by aliasal** [davidedc/Algebrite#91]
At the moment rationalize(1/2*x+3/7) returns the term unchanged.
Numerator gives 1/2*x+3/7 and denominator 1

I'd expect/like to get (7*x+6)/14, 7*x+6 and 14.

I guess this behavior is expected, but is there a way to change it, so that I get the results I'd like to have (or an other option)?

Thanks

**Comment by murkle** [comment]
Try factor() instead

**Comment by davidedc** [comment]
indeed as @murkle says,

  ```numerator(factor(1/2*x + 3/7))```
  ```denominator(factor(1/2*x + 3/7))```

work (both on website and head of master).

However rationalize and numerator or denominator _without_ factor should also work I think, let me take a look.

todo: check the tests to see what kind of behaviour we expect now from rationalize/numerator/denominator and see whether this example should also work

**Comment by aliasal** [comment]
Thanks for your fast replies.

Strangely it does not work for me. At least on my home installation.
On the website (http://algebrite.org/) it does not work when I try the second box (// standard javascript via API) like this:
If I put in 
Algebrite.factor('1/2x+3/7');
I get
1/14*(7*x+6)
Algebrite.numerator('1/14*(7*x+6)');
I get
1/2*x+3/7
or (that is the way I use it in my home installation)
Algebrite.numerator(Algebrite.factor('1/2x+3/7'));
1/2*x+3/7

But in the Fractions box
numerator(factor('1/2x+3/7'));
7*x+6
it works ??

Obviously there is some basic understanding that I am missing.

**Comment by davidedc** [comment]
@aliasal yes so this requires couple of clarifications, and probably a separate github issue.
1. all boxes (and the sandbox) behave the same in respect to all the points here - any difference might be the result of confusion in the copy-paste
2. "Javascript" mode can be used in any box, and in the sandbox, and in the browser console. In the boxes and in the sandbox, we determine whether the input is JS based on the simple heuristic of whether a semicolon appears in the input.
3. ```numerator(factor('1/2x+3/7'));``` actually gives an error, because the semicolon trips the heuristic, and is treated like JS but in the JS environment those functions don't exist in that form
4. ```Algebrite.numerator(Algebrite.factor('1/2x+3/7'));``` fails because when invoked from JS that way, ```Algebrite.numerator``` evaluates the result of ```Algebrite.factor('1/2x+3/7')```, hence causing an expansion and undoing what "factor" just did. This might be a separate issue that needs looking into because it's unclear to me when this further evaluation would be useful.

If you want to run via JS (in any box or sandbox or browser console), you can use the following:
    ```Algebrite.run('numerator(factor(1/2*x + 3/7))');```
    ```Algebrite.run('denominator(factor(1/2*x + 3/7))');```

In these cases, the ";" will indicate JS mode, and "Algebrite.run" will take care of doing the whole thing - and in this case "numerator" will evaluate its argument once, so it won't expand the factored input, so it works.

**Comment by aliasal** [comment]
Thanks, that clarified the issue.
"numerator(factor('1/2x+3/7'));" was indeed a copy-pasty typo. I used it without the semicolon in the sandbox
Algebrite.run does the trick, thanks very much

**Comment by davidedc** [comment]
re-opening, because the workaround is fine, but ideally rationalize/numerator/denominator should work without explicitly needing "factor"
